src/framework/utils.hpp

```diff
@@ -110,6 +110,16 @@ template <class T> matrix<T> partial_trace_b(const matrix<T> &rho, size_t dimB);
 // Tensor product
 template <class T> matrix<T> tensor_product(const matrix<T> &A, const matrix<T> &B);
 
+// Matrix concatenate and split.
+template <class T> matrix<T> concatenate (const matrix<T> &A, const matrix<T> &B, uint_t axis);
```

Comment:
Can you add the documentation about axis = 0 or 1 here for how the concatenation works, same for the `split` function

Reply:
I moved the documentation from the implementation to the header and improved it.